test/test_handlers.py

```diff
@@ -58,3 +58,61 @@ def model():
 
     model = handlers.condition(model, {'x': 1., 'y': 2.})
     assert model() == 3.
+
+
+def model_nested_plates_0():
+    with numpyro.plate('outer', 10):
+        x = numpyro.sample('y', dist.Normal(0., 1.))
+        assert x.shape == (10,)
+        with numpyro.plate('inner', 5):
+            y = numpyro.sample('x', dist.Normal(0., 1.))
+            assert y.shape == (5, 10)
+
+
+def model_nested_plates_1():
+    with numpyro.plate('outer', 10, dim=-2):
+        x = numpyro.sample('y', dist.Normal(0., 1.))
+        assert x.shape == (10, 1)
+        with numpyro.plate('inner', 5):
+            y = numpyro.sample('x', dist.Normal(0., 1.))
+            assert y.shape == (10, 5)
+
+
+def model_nested_plates_2():
+    outer = numpyro.plate('outer', 10)
+    inner = numpyro.plate('inner', 5, dim=-3)
+    with outer:
+        x = numpyro.sample('x', dist.Normal(0., 1.))
+        assert x.shape == (10,)
+    with inner:
+        y = numpyro.sample('y', dist.Normal(0., 1.))
+        assert y.shape == (5, 1, 1)
+
+    with outer, inner:
+        xy = numpyro.sample('xy', dist.Normal(0., 1.), sample_shape=(10,))
+        assert xy.shape == (5, 1, 10)
+
+
+def model_subsample_1():
+    outer = numpyro.plate('outer', 20, subsample_size=10)
+    inner = numpyro.plate('inner', 10, subsample_size=5, dim=-3)
+    with outer:
+        x = numpyro.sample('x', dist.Normal(0., 1.))
+        assert x.shape == (10,)
+    with inner:
+        y = numpyro.sample('y', dist.Normal(0., 1.))
+        assert y.shape == (5, 1, 1)
+
+    with outer, inner:
+        xy = numpyro.sample('xy', dist.Normal(0., 1.))
+        assert xy.shape == (5, 1, 10)
+
+
+@pytest.mark.parametrize('model', [
+    model_nested_plates_0,
+    model_nested_plates_1,
+    model_nested_plates_2,
+    model_subsample_1,
+])
+def test_plate(model):
+    jit(handlers.seed(model, random.PRNGKey(1)))
```

Comment:
nit: I think you need to call this function.

Reply:
To be safe, could you return all sample sites in all models and add a statement to check if the output of jitted model is the same as non-jitted one?

Reply:
Done; just checking the trace values of sample sites.